assets/js/modules/analytics/datastore/properties.js

```diff
@@ -138,49 +138,50 @@ const baseActions = {
+	selectProperty( propertyID, internalPropertyID = '' ) {
 		invariant( isValidPropertySelection( propertyID ), 'A valid propertyID selection is required.' );
 
-		const registry = yield Data.commonActions.getRegistry();
-
-		const accountID = registry.select( STORE_NAME ).getAccountID();
-		if ( ! isValidAccountID( accountID ) ) {
-			return;
-		}
+		return ( function* () {
+			const registry = yield Data.commonActions.getRegistry();
 
-		registry.dispatch( STORE_NAME ).setPropertyID( propertyID );
-
-		if ( PROPERTY_CREATE === propertyID ) {
-			registry.dispatch( STORE_NAME ).setProfileID( PROFILE_CREATE );
-			return;
-		}
+			const accountID = registry.select( STORE_NAME ).getAccountID();
+			if ( ! isValidAccountID( accountID ) ) {
+				return;
+			}
 
-		yield baseActions.waitForProperties( accountID );
-		const property = registry.select( STORE_NAME ).getPropertyByID( propertyID ) || {};
+			registry.dispatch( STORE_NAME ).setPropertyID( propertyID );
 
-		if ( ! internalPropertyID ) {
-			internalPropertyID = property.internalWebPropertyId;
-		}
+			if ( PROPERTY_CREATE === propertyID ) {
+				registry.dispatch( STORE_NAME ).setProfileID( PROFILE_CREATE );
+				return;
+			}
 
-		registry.dispatch( STORE_NAME ).setInternalWebPropertyID( internalPropertyID || '' );
+			yield baseActions.waitForProperties( accountID );
+			const property = registry.select( STORE_NAME ).getPropertyByID( propertyID ) || {};
 
-		if ( property.defaultProfileId ) {
-			registry.dispatch( STORE_NAME ).setProfileID( property.defaultProfileId ); // Capitalization rule exception: defaultProfileId
-			return;
-		}
+			if ( ! internalPropertyID ) {
+				internalPropertyID = property.internalWebPropertyId;
+			}
 
-		// Clear any profile ID selection in the case that selection falls to the getProfiles resolver.
-		registry.dispatch( STORE_NAME ).setProfileID( '' );
+			registry.dispatch( STORE_NAME ).setInternalWebPropertyID( internalPropertyID || '' );
 
-		const profiles = registry.select( STORE_NAME ).getProfiles( accountID, propertyID );
-		if ( profiles === undefined ) {
-			return; // Selection will happen in in getProfiles resolver.
-		}
+			const profiles = registry.select( STORE_NAME ).getProfiles( accountID, propertyID );
+			if ( property.defaultProfileId && profiles?.some( ( profile ) => profile.id === property.defaultProfileId ) ) {
+				registry.dispatch( STORE_NAME ).setProfileID( property.defaultProfileId ); // Capitalization rule exception: defaultProfileId
+				return;
+			}
 
-		const matchedProfile = profiles.find( ( { webPropertyId } ) => webPropertyId === propertyID ) || { id: PROFILE_CREATE }; // Capitalization rule exception: webPropertyId
+			// Clear any profile ID selection in the case that selection falls to the getProfiles resolver.
+			registry.dispatch( STORE_NAME ).setProfileID( '' );
+			if ( profiles === undefined ) {
+				return; // Selection will happen in in getProfiles resolver.
+			}
```

Comment:
I think this should probably still happen before `getProfiles` is called to ensure that clearing the ID happens before the resolver runs.